### ISSUE TITLE
XML response is clipped at certain value, and parsing fails

### ISSUE THREAD

**Issue by rvypandey** [wso2-extensions/siddhi-io-http#88]
**Description:**
When a big XML response is received by @source(type='http-response'...), the response is clipped at certain length, and then the clipped response is tried to be parsed (which fails)

**Suggested Labels:**
bug

**Suggested Assignees:**
@minudika 

**Affected Product Version:**
- siddhi-io-http-1.0.39
- latest SP

**Steps to reproduce:**
- Create a http service that responds with a big XML response. (In my case, big was ~1300 character long w/o whitespaces)
- Setup http-request / http-response. I have it like this:
```
--Sink stream - get http events
@sink(type = 'http-request', publisher.url = 'http://localhost:3000/lastevent',
	sink.id = 'request-sink', 
	@map(type = 'json'))
define stream postHttpRequestStream(payloadBody String); -- payloadBody is currently not used

--Source stream - incoming http events
@source(type = 'http-response', sink.id = 'request-sink', http.status.code = '200', 
	@map(type = 'xml', 
		@attributes(eventName = 'reply/equipmentReplies/movementEvents/sightingEventCodeText')))
define stream httpResponseStream2xx(eventName string);
```

The response from above URL - localhost:3000/lastevent, in postman is:
![postman response](https://user-images.githubusercontent.com/25342287/47149290-c5f5bb80-d2f0-11e8-8509-006d035aae2a.PNG)

- Run this code in stream processor, I get an error-

`[2018-10-18_15-51-26_165] INFO {org.wso2.extension.siddhi.map.xml.sourcemapper.XmlSourceMapper} - Error parsing incoming XML event : trainIIAlphaCode>ICHR</trainIIAlphaCode><sightingEventCode>4050</sightingEventCode><sightingEventCodeText>JUNCTION RECEIVED</sightingEventCodeText><toMarkId>BNSF</toMarkId><trainInd>UP</trainInd></movementEvents><tripPlan><tripPlanId/></tripPlan><waybill><commodity/><destination><city/><splc/><state/></destination><grossWeight>0</grossWeight><netWeight>0</netWeight><tareWeight>0</tareWeight><weightCode>8</weightCode></waybill><requestType>LAST_EVENT</requestType></equipmentReplies></reply></ns2:TracingPlatformRequest>. Reason: javax.xml.stream.XMLStreamException: ParseError at [row,col]:[1,1] Message: Content is not allowed in prolog.. Hence dropping message chunk (Encoded)
`

----
**Additional information:**
If I keep resizing the XML output from server, by removing XML elements, I reach a point where the error is no more.

I have validated my XML via multiple XML validators online. This is the raw version of the XML output:
`<ns2:TracingPlatformRequest xmlns:ns2="http://someUrl/someEndPoint"><TracingPlatformRequestHeader>TracingPlatformRequestHeaderVal</TracingPlatformRequestHeader><reply><equipmentReplies><groupSequenceNumber>0001</groupSequenceNumber><equipment><equipmentId><equipmentInitial>BNSF</equipmentInitial><equipmentNumber>0000000100</equipmentNumber></equipmentId></equipment><umlerDetails><owner>BNSF</owner></umlerDetails><movementEvents><eta><etaDate>Thu Oct 18 2018</etaDate><ediEventCode/><eventCode/><location><splc/></location><reportingScac/></eta><fromMarkId>UP</fromMarkId><loadEmptyStatus>E</loadEmptyStatus><location><splc>536870000</splc></location><reportingRailroadSCAC>BNSF</reportingRailroadSCAC><sightingDate>Thu Oct 18 2018</sightingDate><sightingClmCode>R</sightingClmCode><trainIIAlphaCode>ICHR</trainIIAlphaCode><sightingEventCode>4050</sightingEventCode><sightingEventCodeText>JUNCTION RECEIVED</sightingEventCodeText><toMarkId>BNSF</toMarkId><trainInd>UP</trainInd></movementEvents><tripPlan><tripPlanId/></tripPlan><waybill><commodity/><destination><city/><splc/><state/></destination><grossWeight>0</grossWeight><netWeight>0</netWeight><tareWeight>0</tareWeight><weightCode>8</weightCode></waybill><requestType>LAST_EVENT</requestType></equipmentReplies></reply></ns2:TracingPlatformRequest>`

Thanks,
Ravi Pandey

**Comment by rvypandey** [comment]
Hi @minudika. Just wanted to check if you could reproduce this, and if there is any update on this issue.

Thanks,
Ravi Pandey

**Comment by minudika** [comment]
Hi @rvypandey,

We have observed this [issue](https://github.com/wso2-extensions/siddhi-io-http/issues/84) and already fixed this on the latest siddhi-io-http version.
In order to apply the latest version on your Stream Processor pack, please follow the below steps.

1. Download **siddhi-io-http-1.0.40.jar** from the following link.
https://store.wso2.com/store/assets/analyticsextension/details/63a5db75-586e-48d0-8244-4c16774e8846
2. **Replace** the existing **siddhi-io-http-1.0.39.jar** which can be found in the  **\<SP-HOME\>/lib** directory with the latest **siddhi-io-http-1.0.40.jar**.

Please let us know if you have further concerns regarding this.

Regards,
Minudika

**Comment by rvypandey** [comment]
Thank you @minudika,

I have verified the fix and I cannot reproduce this in 1.0.40.

Regards,
Ravi Pandey
